images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -247,48 +227,47 @@ vrrp_instance ${instance_name} {
 #      generate_failover_configuration
 #
 function generate_failover_config() {
-  local vips=$(expand_ip_ranges "$HA_VIPS")
-  local interface=$(get_network_device "$NETWORK_INTERFACE")
-  local ipaddr=$(get_device_ip_address "$interface")
-  local port=$(echo "$HA_MONITOR_PORT" | sed 's/[^0-9]//g')
+  local vips=$(expand_ip_ranges "${HA_VIPS}")
+  local interface=$(get_network_device "${NETWORK_INTERFACE}")
+  local ipaddr=$(get_device_ip_address "${interface}")
+  local port=$(echo "${HA_MONITOR_PORT}" | sed 's/[^0-9]//g')
 
   echo "! Configuration File for keepalived
 
-$(generate_global_config "$HA_CONFIG_NAME")
-$(generate_script_config "$ipaddr" "$port")
-$(generate_vrrp_sync_groups "$HA_CONFIG_NAME" "$HA_VIPS")
+$(generate_global_config "${HA_CONFIG_NAME}")
+$(generate_script_config "${ipaddr}" "${port}")
 "
 
-  local ipkey=$(echo "$ipaddr" | cut -f 4 -d '.')
+  local ipkey=$(echo "${ipaddr}" | cut -f 4 -d '.')
   local ipslot=$((ipkey % 128))
 
-  local nodecount=$(($HA_REPLICA_COUNT > 0 ? $HA_REPLICA_COUNT : 1))
-  local idx=$((ipslot % $nodecount))
+  local nodecount=$((${HA_REPLICA_COUNT} > 0 ? ${HA_REPLICA_COUNT} : 1))
```

Comment:
Do not need `${}` inside of `$(( ))` -- just use `$(( varname ))`.

But in this case, just use a normal `if`, Bash syntax is hard enough as it is without using obscure features like this.

```sh
local nodecount
if [[ "${HA_REPLICA_COUNT}" -gt 0 ]]; then
    nodecount="${HA_REPLICA_COUNT}"
else
    nodecount="1"
fi
```